pkg/reconciler/source/config_watcher.go

```diff
@@ -20,143 +20,263 @@ import (
 	"context"
 
 	"go.uber.org/zap"
+
 	corev1 "k8s.io/api/core/v1"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"knative.dev/eventing/pkg/logging"
 	"knative.dev/pkg/configmap"
-	pkgLogging "knative.dev/pkg/logging"
+	pkglogging "knative.dev/pkg/logging"
 	"knative.dev/pkg/metrics"
 	tracingconfig "knative.dev/pkg/tracing/config"
+)
 
-	"knative.dev/eventing/pkg/logging"
+const (
+	EnvLoggingCfg = "K_LOGGING_CONFIG"
+	EnvMetricsCfg = "K_METRICS_CONFIG"
+	EnvTracingCfg = "K_TRACING_CONFIG"
 )
 
+type ConfigAccessor interface {
+	ToEnvVars() []corev1.EnvVar
+	LoggingConfigEnvVar() corev1.EnvVar
+	MetricsConfigEnvVar() corev1.EnvVar
+	TracingConfigEnvVar() corev1.EnvVar
+}
+
+var _ ConfigAccessor = (*ConfigWatcher)(nil)
+
+// ConfigWatcher keeps track of logging, metrics and tracing configurations by
+// watching corresponding ConfigMaps.
 type ConfigWatcher struct {
```

Comment:
IMO this implementation should be private

Reply:
That's fine in this case you're right, since the type doesn't have type-specific methods that could be useful to a consumer. I was just blindly applying the idiom
> accept interfaces return structs

Reply:
Actually no, that would break `WithLogging`, `WithMetrics`,  `WithTracing`, so it wouldn't be a good pattern here.

Reply:
Ok